src/shell.ts

```diff
@@ -260,23 +292,14 @@ export class Shell {
             { prompt: "Close" }
         ]
 
-        let res
-        try {
-            res = await this.runForm(reconnectForm, "menu")
-        } catch(e) {
-            // try to connect
-            res = null
-        }
-        // TODO: needs refactoring
-        if (res == "Close") {
-            this.map.showLog(false)
-            gate.close()
-            return
-        }
         if (gate.session) {
             gate.session.close()
             gate.session = null
         }
+        let res = await this.runForm(reconnectForm, "menu")
+        // TODO: needs refactoring
+        if (res == "Close")
+            throw new Error("Aborted")
```

Comment:
don't throw an error and push handling back to the gate. Just close the gate here